### PR TITLE
Template transition matrices

### PR DESCRIPTION
**Pull request by asylves1**
# Description

Design Change:

Add dropdown options for the template matrix.
image

Testing: 
- Open a model that has be stratified. 
- Open the template matrix in the diagram, and under the Transition section.
- Both should have the options to show 4 different matrices 

image
image

Resolves #(issue)

### CODE REVIEW

File: packages/client/hmi-client/src/model-representation/mira/mira.ts [hunk 1]
```diff
@@ -345,14 +345,92 @@ export const createInitialMatrix = (miraModel: MiraModel, key: string) => {
...
+		paramValueMap.set(paramObj.name, paramObj.value);
+	});
+
+	// Find templates with expressions that contains one or more of the params
+	const templates = miraModel.templates.filter((t) => {
+		const miraTemplateParam = miraTemplateParams[t.name];
+		const intersection = _.intersection(childrenParams, [miraTemplateParam.name]);
+		return intersection.length > 0;
+	});
+	const subjectControllers = extractSubjectControllersMatrix(
```
Comment (asylves1) [inline packages/client/hmi-client/src/model-representation/mira/mira.ts:389]: Will be enabled in the next matrix PR

File: packages/client/hmi-client/src/model-representation/mira/mira.ts [hunk 2]
```diff
@@ -345,14 +345,92 @@ export const createInitialMatrix = (miraModel: MiraModel, key: string) => {
...
+		templates,
+		childrenParams,
+		paramValueMap,
+		paramLocationMap
+	);
+	const outcomeControllers = extractOutcomeControllersMatrix(
```
Comment (asylves1) [inline packages/client/hmi-client/src/model-representation/mira/mira.ts:395]: Will be enabled in the next matrix PR

File: packages/client/hmi-client/src/model-representation/mira/mira.ts [hunk 3]
```diff
@@ -345,14 +345,92 @@ export const createInitialMatrix = (miraModel: MiraModel, key: string) => {
 	return m2;
 };
 
+/** */
```
Comment (YohannParis) [inline packages/client/hmi-client/src/model-representation/mira/mira.ts:348]: Missing comment documentation here

File: packages/client/hmi-client/src/model-representation/mira/mira.ts [hunk 4]
```diff
@@ -345,6 +345,75 @@ export const createInitialMatrix = (miraModel: MiraModel, key: string) => {
...
+			col: 0,
+			rowCriteria: name,
+			colCriteria: '',
+			content: {
+				id: name,
+				value: miraModel.templates[name]
+			}
+		});
+		other.push(row);
+	});
```
Comment (mwdchang) [inline packages/client/hmi-client/src/model-representation/mira/mira.ts:408]: Calculation of 'others' seems slightly odd as it was an escape-hatch we made specifically for parameters, note templates - which should have regular-looking matrices. 

Can we provide some additional background here on what this task is about?

**Comment by mwdchang** [comment]
I suspect this may not be needed because of fixes in:
- https://github.com/DARPA-ASKEM/terarium/pull/5160, and
- https://github.com/DARPA-ASKEM/terarium/pull/5157

That address template parsing directly upstream of displaying the matrix modal. But let's keep this PR open for a few more days to confirm @asylves1

**Comment by asylves1** [comment]
No longer needed

**Pull request closed without merging**